### DISCUSSION TITLE
Anyone tested 1.28 yet?

### DISCUSSION THREAD

**Discussion by tschuering** [mysticaltech/terraform-hcloud-kube-hetzner#1093] [Q&A]
Hey, currently 1.28 is forbidden as value for k3s_initial_channel, so is this still not working or did anybody already some tests? Otherwise I would suggest a PR to change this. @mysticaltech

**Comment by mysticaltech** [comment]
How come, try "v1.28",  and if not in the allowed values, PR most welcome 🙏

**Comment by mysticaltech** [comment]
It should work, but probably not with Cilium and Calico, as they don't support it yet to the best of my knowledge.

**Reply by maggie44** [reply]
Was this ever merged? https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/pull/995

**Reply by maggie44** [reply]
I think they do support it now. 

https://github.com/cilium/cilium/pull/29004
https://docs.tigera.io/calico/latest/getting-started/kubernetes/requirements#kubernetes-requirements

**Reply by mysticaltech** [reply]
Thanks for the info @maggie44, will update it tonight.

**Reply by M4t7e** [reply]
I don't think they support it in one of the 1.14.* releases. Just because something was merged to the main branch does not mean it is released officially and even if it is, it does also not say in which version either. As far as I can see the changes found the way only to the 1.15.*-pre releases.

Example:
- 1.28 support is in 1.15.0-pre: https://github.com/cilium/cilium/blob/v1.15.0-pre.0/.github/actions/ginkgo/main-k8s-versions.yaml#L4
- 1.28 support is **not** in 1.14.4: https://github.com/cilium/cilium/blob/v1.14.4/.github/actions/ginkgo/main-k8s-versions.yaml#L4

AFAIK there is no plan to support that in any 1.14.* release and it will come in version 1.15 like I mentioned here: https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/pull/995#issuecomment-1742871512

It is relatively easy to check the supported K8s version for different Cilium versions here: https://github.com/cilium/cilium/blob/v1.14.4/Documentation/network/kubernetes/compatibility.rst
Just change the tag to see what is supported and what not

**Comment by Silvest89** [comment]
I've been running 1.28 for ages with cilium 1.14, os and kube auto update enabled. I am currently on 1.28.3 even.
I have it at latest ;)

**Reply by mysticaltech** [reply]
Good will update tonight tonight.

**Comment by mysticaltech** [comment]
@tschuering  @maggie44  v1.28 is now default in v2.10.0.
